File: sub-model/src/main/java/org/github/base/entity/SysRoleMenuEntity.java
package org.github.base.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.IdType;
import org.github.base.AbstractEntity;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.*;

/**
 * <p>
 * 角色和菜单关联表
 * </p>
 *
 * @author JYD_XL
 * @since 2019-12-12
 */
@Setter
@Getter
@EqualsAndHashCode(callSuper = true)
@TableName("sys_role_menu")
public class SysRoleMenuEntity extends AbstractEntity {
    /** UID */
    private static final long serialVersionUID = 1L;

    /** 角色ID */
    @TableId(value = "role_id", type = IdType.ASSIGN_UUID)
    private Long roleId;

    /** 菜单ID */
    @TableField("menu_id")
    private Long menuId;
}

File: sub-model/src/main/java/org/github/base/entity/SysRoleMenuEntity.kt
package org.github.base.entity

import com.baomidou.mybatisplus.annotation.TableName
import org.github.base.AbstractEntity

/**
 * <p>
 * 角色和菜单关联表
 * </p>
 *
 * @author JYD_XL
 * @since 2019-11-19
 */
@TableName("sys_role_menu")
class SysRoleMenuEntity: AbstractEntity() {

    /** 角色ID */
    var roleId: Long? = null

    /** 菜单ID */
    var menuId: Long? = null

}
